src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -55,7 +55,7 @@ public Climber() {
      * @return true if setting speed was successful
      */
     public boolean setSpeed(double spd) {
-        if (spd > 0 && ratchetEngaged) return false;
+        if (spd > 0 && (ratchetEngaged || climberRight.getEncoder().getPosition() > 80 || climberLeft.getEncoder().getPosition() > 80)) return false;
```

Comment:
if (spd >0 && ratchEngaged) return false; // This is needed always, cannot raise when ratchets are engaged

// Now we check motor limits
if (spd > 0) {
  if (climberRight.getEncoder().getPosition() > 80) {
    climberRight.set(0);
  } else {
    climberRight.set(-spd);
  }
  if (climberLeft.getEncoder().getPosition() > 80) {
    climberLeft.set(0);
  } else {
    climberLeft.set(spd);
  }
}

Reply:
Also the right climber (because it's moving negatively), will need the check to be:

if (climberRight.getEncoder().getPosition() < -80) {